File: Ble_v_1.0.0/app/src/main/java/com/ble/application/IApplication.java
package com.ble.application;

import android.app.Application;
import android.content.Context;
import com.sdk.application.SdkApplication;
/**
 * Created by E on 2017/6/7.
 */
public class IApplication extends Application {

    private static Context context = null;

    @Override
    public void onCreate() {
        super.onCreate();

        context = this;

        SdkApplication.init(this);
    }

    public static Context getContext(){
        return context;
    }
}

File: app/src/main/java/com/bluetooth/demo/application/IApplication.kt
package com.bluetooth.demo.application

import android.app.Application
import com.bluetooth.demo.ex.HuiDongBleGattServiceListener
import com.bluetooth.lib.imp.conn.BleCallbackMgr
import com.bluetooth.lib.invoke.BleSdkInitHelper
import com.bluetooth.lib.util.BleSdkContext

class IApplication : Application() {

    override fun onCreate() {
        super.onCreate()
        BleSdkContext.needCheckLocationPermission = true
        BleSdkInitHelper.initSDK(this)

        //初始化
        BleCallbackMgr.registerGattServiceListener(HuiDongBleGattServiceListener())
    }

}